lib/mapObjects/CRewardableConstructor.cpp

```diff
@@ -132,6 +132,7 @@ void CRandomRewardObjectInfo::configureReward(CRewardableObject * object, CRando
 	reward.artifacts = JsonRandom::loadArtifacts(source["artifacts"], rng);
 	reward.spells = JsonRandom::loadSpells(source["spells"], rng, spells);
 	reward.creatures = JsonRandom::loadCreatures(source["creatures"], rng);
+	reward.casts = JsonRandom::loadSpells(source["casts"], rng, spells);
```

Comment:
Perhaps load only single spell (I doubt that TP + DD in the same object will work properly anyway)
And also load mastery level (none/basic/advanced/expert)

Reply:
agree, also thinking on.